# awesome-single-cell-foundation

## Why creating a repo focusing on single cell foundation models?

The evolution of single-cell biology is intersecting with the advancements of artificial intelligence and machine learning. The rise of foundation models, especially large language networks, suggests potential deeper insights into cellular mechanisms. Anticipating a future where a foundational cell model integrates molecular, spatial, and morphometric details offers a promising avenue for holistic biological understanding. This convergence of technology and biology brings forth exciting opportunities to enhance our knowledge of cellular states and their implications in health and disease.

**Quote from Fabian Theis:** As modern language models enable text generation as answer to a query, we will be able to query a cellular foundation model about unseen, new cell states across health and disease.



## Single cell RNA sequencing (scRNA-seq)

- GeneCompass: Deciphering Universal Gene Regulatory Mechanisms with Knowledge-Informed Cross-Species Foundation Model [paper](https://www.biorxiv.org/content/10.1101/2023.09.26.559542v1), [code](https://github.com/xCompass-AI/GeneCompass)

- Geneformer: Transfer learning enables predictions in network biology [paper](https://www.nature.com/articles/s41586-023-06139-9), [code](https://huggingface.co/ctheodoris/Geneformer)

- scGPT: Towards Building a Foundation Model for Single-Cell Multi-omics Using Generative AI [paper](https://www.biorxiv.org/content/10.1101/2023.04.30.538439v1), [code](https://github.com/bowang-lab/scGPT)

- scBert: scBERT as a large-scale pretrained deep language model for cell type annotation of single-cell RNA-seq data [paper](https://www.nature.com/articles/s42256-022-00534-z), [code](https://github.com/TencentAILabHealthcare/scBERT)

- scFoundation: Large Scale Foundation Model on Single-cell Transcriptomics [paper](https://www.biorxiv.org/content/10.1101/2023.05.29.542705v3), [code](https://github.com/biomap-research/scFoundation)

- tGPT: Generative pretraining from large-scale transcriptomes for single-cell deciphering [paper](https://www.sciencedirect.com/science/article/pii/S2589004223006132), [code](https://github.com/deeplearningplus/tGPT)

- CellLM: Large-Scale Cell Representation Learning via Divide-and-Conquer Contrastive Learning [paper](https://arxiv.org/pdf/2306.04371.pdf), [code](https://github.com/PharMolix/OpenBioMed/blob/main/README.md)

- Exceiver: A single-cell gene expression language model [paper](https://arxiv.org/pdf/2210.14330.pdf), [code](https://github.com/keiserlab/exceiver)

- scTranslator: Training on large-scale RNA data to impute protein abundance [paper](https://t.co/DHRtCmzaGK), [code](https://t.co/TC0OCOc0q7)


## Spatially-resolved Transcriptomic (SRT)

- CellPLM: Pre-training of Cell Language Model Beyond Single Cells [paper](https://www.biorxiv.org/content/10.1101/2023.10.03.560734v1.full.pdf), [code](https://github.com/OmicsML/CellPLM)


## Paired  scATAC-seq and scRNA-seq

- GET: a foundation model of transcription across human cell types [paper](https://www.biorxiv.org/content/10.1101/2023.09.24.559168v1.full), [code](https://github.com/GET-Foundation)



## Incorporating Text

- BioTranslator: Multilingual translation for zero-shot biomedical classification using BioTranslator [paper](https://www.nature.com/articles/s41467-023-36476-2), [code](https://github.com/HanwenXuTHU/BioTranslatorProject)


# Benchmarking

- Assessing the limits of zero-shot foundation models in single-cell biology [paper](https://www.biorxiv.org/content/10.1101/2023.10.16.561085v1.full.pdf), [code](https://github.com/microsoft/zero-shot-scfoundation)